src/directive.js

```diff
@@ -33,20 +37,39 @@
                         throwNoKeyException();
                     }
 
-                    var callback = function () {
-                        // Notify about the response availability
-                        scope.onSuccess({response: vcRecaptcha.getResponse(scope.widgetId)});
+                    var callback = function (gRecaptchaResponse) {
+                        // Safe $apply
+                        $timeout(function () {
+                            if(ctrl){
+                                ctrl.$setValidity('recaptcha',true);
+                            }
+                            scope.response = gRecaptchaResponse;
+                            // Notify about the response availability
+                            scope.onSuccess({response: gRecaptchaResponse, widgetId: scope.widgetId});
+                        });
+
+                        // captcha session lasts 2 mins after set.
+                        $timeout(function (){
+                            if(ctrl){
+                                ctrl.$setValidity('recaptcha',false);
+                            }
+                            scope.response = "";
+                            // Notify about the response availability
+                            scope.onExpire({widgetId: scope.widgetId});
+                        }, 2 * 60 * 1000);
                     };
 
-                    vcRecaptcha.create(elm[0], scope.key, callback, {
+                    vcRecaptcha.create(elm[0], key, callback, {
 
-                        theme: attrs.theme || null
+                        theme: scope.theme || null
```

Comment:
My mistake! I missed [this change](https://github.com/TheSharpieOne/angular-recaptcha/blob/cffdd98484078638ecceac165e10eb8a4915e8be/src/directive.js#L19). Disregard my comment above.